web-client/src/lib.rs

```diff
@@ -343,6 +346,43 @@ impl WebClient {
 
         Ok(())
     }
+
+    /// This function is used to query the network for transactions from some specific adress, that
+    /// have been included in the chain.
+    /// The obtained transactions correspond to extended transactions.
+    /// They are verified before being returned.
+    #[wasm_bindgen(js_name = requestTransactionsFromAddress)]
+    pub async fn request_transations_from_address(
+        &self,
+        address: Address,
+        max: u16,
+    ) -> Result<Uint8Array, JsError> {
+        let transactions = self
+            .inner
+            .consensus_proxy()
+            .request_transactions_by_address(address.get_native(), Some(max))
+            .await?;
+
+        let executed_txs: Vec<ExecutedTransaction> = transactions
+            .into_iter()
+            .map(|ext_transaction| ext_transaction.into_transaction().unwrap())
+            .collect();
+
+        // TODO: We are converting executed transactions into a regular transaction, which loses the execution result of the txn
+        let js_transactions: Vec<Transaction> = executed_txs
+            .into_iter()
+            .map(|executed_tx| Transaction::from_native(executed_tx.get_raw_transaction().clone()))
+            .collect();
+
+        let mut js_values = vec![];
+
+        for js_transaction in js_transactions {
+            //We need to serialize the exported transaction
+            js_values.extend(js_transaction.serialize());
```

Comment:
Serialized stuff is unfortunately very useless on the JS side, if we don't also ship an implementation in JS that can unserialize it.

So I would think how we can best return an array of structs here.

Reply:
This issue mentions several ways to pass arrays of custom structs across to JS. I think the most promising and least memory intensive (we don't want to keep up to 500 tx structs in wasm memory, do we?) is to convert to JSON or at least [serialize with serde.](https://github.com/cloudflare/serde-wasm-bindgen)

https://github.com/rustwasm/wasm-bindgen/issues/111

Reply:
Yes we are exploring different ideas, but I think serializing to JSON seems the most suitable option